src/process-events.c

```diff
@@ -444,12 +447,10 @@ int BPF_KPROBE_SYSCALL(kprobe__sys_exec_tc_argv,
         goto Tail;
     }
 
-    long count = 0;
-    u32 ii = 0;
-    // explicit copy from pointer to stack, satisfy verifier
-    __builtin_memcpy(&ii, pii, sizeof(u32));
-    u64 ptr = 0;
-    u64 ret = 0;
+    int count = 0;
+    u32 ii = pii[0];
```

Comment:
Not sure why you're doing `pii[0]` instead of `*pii` here?  They should do the same thing, but using the array syntax suggests that there's an array involved.

Reply:
these are stored in a single element array specified at the start of the file:

```c
struct bpf_map_def SEC("maps/read_flush_index") read_flush_index = {
    .type = BPF_MAP_TYPE_PERCPU_ARRAY,
    .key_size = sizeof(u32),
    .value_size = sizeof(u32),
    .max_entries = 1,
    .pinning = 0,
    .namespace = "",
};
```

This is an array per cpu, where each array holds at most 1 element of key and value sized to a `u32`. 

I figured this would make it more obvious that an array is somehow involved as you said, but it should be equivalent.

Reply:
Ah, ok, but I think `bpf_map_lookup_elem` is already doing the array indexing and is just returning a pointer to the single indexed element.  This runaround is apparently so the lookup function is generic across different map types; e.g. one of the hashmap-based maps wouldn't be just a straightforward array index.

In other words, the pointer returned is conceptually some single element from somewhere in a map, and not the first of an array of values, even if it actually is in this case the first of a single-element array.

Reply:
Ah you are right! I will make that be just a dereference then!